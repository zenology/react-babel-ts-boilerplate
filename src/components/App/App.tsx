import React from 'react'

interface AppProps {
  name?: string
}

const App: React.SFC<AppProps> = (props) => {
  return (
    <h1>
      Hello World! <span css={{ color: 'red' }}>{props.name}</span>
    </h1>
  )
}

App.defaultProps = {
  name: 'Nom Nom Nom',
}

export default App
