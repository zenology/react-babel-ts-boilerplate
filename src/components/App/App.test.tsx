import React from 'react'
import { matchers } from 'jest-emotion'
import { render } from '@testing-library/react'

import App from './App'

expect.extend(matchers)

describe('App', () => {
  it('Should render properly', () => {
    const { container } = render(<App />)

    expect(container).toMatchSnapshot()
  })

  it('Show correct name from props', () => {
    const name = 'Noppanan Sinroungrong'

    const { findByText } = render(<App name={name} />)

    expect(findByText(name)).toBeTruthy()
  })

  it('Should display color on name correctly', () => {
    const { container } = render(<App />)
    const spanTag = container.querySelector('span')

    expect(spanTag).toHaveStyleRule('color', 'red')
  })
})
