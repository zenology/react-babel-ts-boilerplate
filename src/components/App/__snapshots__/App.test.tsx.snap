// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render properly 1`] = `
.emotion-0 {
  color: red;
}

<div>
  <h1>
    Hello World! 
    <span
      class="emotion-0"
    >
      Nom Nom Nom
    </span>
  </h1>
</div>
`;
